### PR TITLE
CI: Check if CODEOWNERS reflect chart maintainers

### PR DESCRIPTION
**Pull request by torstenwalter**
#### What this PR does / why we need it:

Script generates CODEOWNERS based on chart maintainers and checks if generated file shows a diff.

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed

### CODE REVIEW

**Comment by torstenwalter** [comment]
Should be merged after #275

File: .github/workflows/check-codeowners.yaml
Comment (monotek) [inline .github/workflows/check-codeowners.yaml:11]: not sure how snap handles majo apllication updates inside same os version.
so i would suggest not to use the snap but to install yq via binary download, with a fixed version.

maybe we should also go for a fixed os version instead "ubuntu-latest".

**Review by monotek — CHANGES_REQUESTED** [review]

**Comment by monotek** [comment]
Not stale. #275 is merged so I guess ready to proceed.

File: .github/workflows/check-codeowners.yaml
Comment (scottrigby) [reply .github/workflows/check-codeowners.yaml:11]: Unless we need pinned versions I'd say we keep it simple. 

- For GitHub actions it's safe to use `ubuntu-latest` unless a specific version for an application is required (ref: https://github.com/actions/virtual-environments)
- snap is the preferred method for installing yq on Ubuntu (https://github.com/mikefarah/yq#ubuntu-and-other-linux-distros-supporting-snap-packages). I notice it took about 20s, which isn't too bad for this use (we could wget/curl it, but then we'd incur the maintenance cost of keeping it up to date when the version doesn't matter for our use case). I haven't tested the official GitHub action, but this open issue and comment by the yq maintainer doesn't make me feel too motivated to recommend it yet: https://github.com/mikefarah/yq/issues/418

**Review by scottrigby — APPROVED** [review]
Just needs a `main` merge/rebase to pass the check, otherwise LGTM

**Comment by torstenwalter** [comment]
rebased changes against `main`.

File: .github/workflows/check-codeowners.yaml
Comment (torstenwalter) [reply .github/workflows/check-codeowners.yaml]: It's now using the github action to download yq, which under the hood uses snap to install it ;-)

Comment (scottrigby) [reply .github/workflows/check-codeowners.yaml]: Haha cool

Comment (monotek) [reply .github/workflows/check-codeowners.yaml]: Ok, let's keep the snap.

I already had problems using Ubuntu latest as they just moved from Ubuntu 18.04 to 20.04. problems been Python related. So I guess this could be a problem again when the image switches to Ubuntu 22.04.

**Review by monotek — COMMENTED** [review]
See comment above

**Comment by torstenwalter** [comment]
> See comment above

@monotek What should be changed?

**Review by monotek — APPROVED** [review]

**Review by scottrigby — APPROVED** [review]
👌

**Comment by bismarck** [comment]
@torstenwalter I got this failure after this change was merged: https://github.com/prometheus-community/helm-charts/actions/runs/461677874

**Comment by torstenwalter** [comment]
@bismarck Thanks for the ping. I am looking into it.

**Comment by torstenwalter** [comment]
@bismarck should be fixed now.

**Pull request merged**
